openapi.yaml

```diff
@@ -132,120 +556,313 @@ securityDefinitions:
+              ingredients:
+                properties:
+                  data:
+                    type: array
+                    items:
+                      $ref: '#/definitions/IngredientRelationshipResource'
+  PizzaPatchBody:
+    required: [data]
+    properties:
+      data:
+        required: [id, type]
+        properties:
+          type: *pizzaType
+          id: *pizzaId
+          attributes:
+            $ref: '#/definitions/PizzaAttributes'
+          relationships: *pizzaUpdateRelationships
+  PizzaPostAttributes:
+    type: object
+    required: [name, ovenTemp, bakeTime]
+    properties:
+      name: &pizzaName
+        type: string
+        description: the name of the pizza
+        example: My Favorite Margherita
+      ovenTemp: &ovenTemp
         type: integer
-        description: Total number of results
-        example: 10
-      totalPages:
+        description: the temperature to bake the pizza at, in Fahrenheit
+        example: 500
+      bakeTime: &bakeTime
         type: integer
-        description: Total number of pages
+        description: the time in minutes to bake the pizza for
         example: 10
-      currentPageNumber:
-        type: integer
-        description: Page number of the returned results
-        example: 1
-      currentPageSize:
-        type: integer
-        description: Number of results per page
-        example: 25
-  SelfLink:
-    properties:
-      self:
+      specialInstructions: &pizzaInstructions
         type: string
-        format: url
-        description: Self-link of current resource
-  PaginationLinks:
+        description: any special instructions for baking the pizza
+        example: place the basil on after baking
+        default: ''
+  # distinction from main other attributes: fields are not required
+  PizzaAttributes:
+    type: object
     properties:
-      first:
-        type: string
-        format: url
-        description: The first page of data
-      last:
+      name: *pizzaName
+      ovenTemp: *ovenTemp
+      bakeTime: *bakeTime
+      specialInstructions: *pizzaInstructions
+  IngredientsResult:
+    properties:
+      data:
+        type: array
+        items:
+          $ref: '#/definitions/IngredientResource'
+      links:
+        $ref: '#/definitions/SelfLink'
+  IngredientResult:
+    properties:
+      data:
+        $ref: '#/definitions/IngredientResource'
+      links:
+        $ref: '#/definitions/SelfLink'
+  IngredientAttributes:
+    type: object
+    properties: &ingredientProperties
+      name:
         type: string
-        format: url
-        description: The last page of data
-      prev:
+        description: the name of the ingredient
+        example: Sausage
+      ingredientType:
         type: string
-        format: url
-        description: The previous page of data
-      next:
+        description: the type of the ingredient
+        enum: [meat, vegetable, fruit, other]
```

Comment:
Been a bit since I wrote openapi. I see this enum list both here and above on one of the endpoints. Is there a way to define the enum list and reference it when needed?

Reply:
I don't think you can do it with openAPI but you could use yaml anchors/aliases.